### PR TITLE
Adds a filter to remove __ keys in dictionaries. This is needed to ensure guppy will work with the opensearch indexes

### PR DESCRIPTION
**Pull request by craigrbarnes**
Link to JIRA ticket if there is one: 

### New Features
* adds filter "strip_leading_double_underscore" which will remove __ from dictionary object and replace them with _key.

### CODE REVIEW

**Review by mfshao — DISMISSED** [review]

**Comment by mfshao** [comment]
I guess my point is that considering what we are planning to put into MDS/AggMDS, especially with the idea of adding fine-grained access control, the current behavior of those __ fields might still be useful. It would be great if we can have some documents on how to address this use case and compare with different design solutions, if any

File: src/mds/agg_mds/adapters.py [hunk 1]
```diff
@@ -127,6 +162,7 @@ class FieldFilters:
     filters = {
         "strip_html": strip_html,
         "strip_email": strip_email,
+        "strip_leading_double_underscore": rename_double_underscore_keys,
```
[the remark this answers is not in the document]
Comment (craigrbarnes) [reply src/mds/agg_mds/adapters.py:165]: There are other fields loaded that start with __, other than __manifest, which show up in nested fields. This is to support using aggMDS ETL to load data for use with Guppy without modification. 

 In fact, for M2DC, we are changing `__manitest` to `_manifest`. However, other fields have __ in nested subfields that crash guppy.  The other alternative is:
 
 1. Add support for mapping nested subfields; however, this will result in more complex code that will be harder to maintain. 
 2. Modify guppy to exclude fields. I had both a feature doc and working code; however, this was rejected by the tech leads. 
 
 Consider this for ETL, as it does not affect the API or deployment of any Gen3 service,  I don't see a need for a feature document.

[the remark this answers is not in the document]
Comment (mfshao) [reply src/mds/agg_mds/adapters.py:165]: Yeah I think ultimately I'm concerned about modifying the keys in data during ETL in this sort of implicity ways, the site admin need to be well aware of how `strip_leading_double_underscore` would work and what the expect changes would be. 
And by reading this implementation, I think if there are 3 _ already exists in the field name, the filter will update it to double _ as indicated in the unit tests. I'm not sure if this is an intentional implemtation for a use case, or an newly introduced edge case. If the goal is to modify field names with leading double _ as the name of this filter suggested, would it be better for the filter to only alter field names that has exactly 2 _s in its key?
Second I think we should add description of how this new filter works into https://github.com/uc-cdis/metadata-service/blob/master/docs/config_agg_mds.md#filters

**Review by urvi-occ — COMMENTED** [review]
code looks good but it removes my `WindberSubjectAdapter` code so can you change that?

File: src/mds/agg_mds/adapters.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (mfshao) [reply src/mds/agg_mds/adapters.py:165]: I have no idea about the windber adapter and why it is getting removed from here. I'll defer that to others

**Review by urvi-occ — APPROVED** [review]

**Comment by craigrbarnes** [comment]
> code looks good but it removes my `WindberSubjectAdapter` code so can you change that?

Ugh. Yes I will restore that now.

File: docs/config_agg_mds.md [hunk 2]
```diff
@@ -334,7 +334,38 @@ Example:
...
+Example:
+```json
+"filesCount": {
+          "path": "filesCount",
+          "filters": [
+            "strip_leading_double_underscore"
+          ]
+        },
+```
+Will remove all ```__``` (double underscores) from any child object in the field ``filesCount`. For example:
```
Comment (mfshao) [inline docs/config_agg_mds.md:347]:
```suggestion
Will remove all ```__``` (double underscores) from any child object in the field `filesCount`. For example:
```

File: docs/config_agg_mds.md [hunk 3]
```diff
@@ -563,7 +594,7 @@ this will the look for metadata entries such as:
 ### Advanced filtering
 
 The Gen3 metadata-service supports filtering as described in the documentation. The Gen3 Adapter
-allows a filter option to be configs which is passed to the MDS. Specific studies can
+allows a filter option to be configed which is passed to the MDS. Specific studies can
```
Comment (mfshao) [inline docs/config_agg_mds.md:597]:
```suggestion
allows a filter option to be configured which is passed to the MDS. Specific studies can
```

File: src/mds/agg_mds/adapters.py [hunk 4]
```diff
@@ -66,6 +66,42 @@ def aggregate_pdc_file_count(record: list):
...
+    else:
+        return d
+
+
+def rename_single_dict(d):
+    """Helper function to rename keys in a single dictionary."""
+    result = {}
+    for key, value in d.items():
+        # ensure keys is a string and is only leading with __
+        if isinstance(key, str) and len(key) > 2 and key[:2] == "__" and key[2] != "_":
```
Comment (mfshao) [inline src/mds/agg_mds/adapters.py:95]: Consider using regex match testing instead of this if logic chains, can just check if `key` is string then call re.match() on `key`
I think something like this `^__[^_][\S]+$` should work, but I haven't tested

**Review by m0nhawk — APPROVED** [review]
